### PR TITLE
[Mailer] Add the `allowed_recipients` option

### LINKED ISSUE #19769: [Mailer] Add support for allowing some users even if `recipients` is de…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#54044
| PR author(s) | @lyrixx
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix #19769

### CODE REVIEW

File: mailer.rst [hunk 1]
```diff
@@ -1852,6 +1852,66 @@ a specific address, instead of the *real* address:
...
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <!-- ... -->
+            <framework:config>
+                <framework:mailer>
+                    <framework:envelope>
+                        <framework:recipient>youremail@example.com</framework:recipient>
+                        <framework:allowed-recipient>internal@example.com</framework:recipient>
```
Comment (xabbuh) [inline mailer.rst]:
```suggestion
                        <framework:allowed-recipient>internal@example.com</framework:allowed-recipient>
```

Comment (alexandre-daubois) [reply mailer.rst]: Of course, thanks!

**Comment by xabbuh** [comment]
We should somehow document that the `allowed_recipients` option is used to configure a list of regular expressions. With the current example one can get the expression that I would have to list all recipients explicitly.

**Comment by alexandre-daubois** [comment]
You're right I missed that point! I updated accordingly with an example

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
A very interesting option and a nice contribution 🙌  Thanks Alex!

**Pull request merged**